docs/src/model_creation/reactionsystem_content_accessing.md

```diff
@@ -0,0 +1,249 @@
+```@example model_accessing_basics
+issetequal([sir.S, sir.I, sir.R], species(sir))
+```
+
+If we wish to count the number of parameters or species in a system, we can do this directly through the `numparams` and `numspecies` functions:
+```@example model_accessing_basics
+numparams(sir)
+```
+```@example model_accessing_basics
+numspecies(sir)
+```
+
+### [Accessing model reactions](@id model_accessing_basics_reactions)
+A vector containing all a model's [reactions](@ref ref) can be retrieved using the `reactions` function:
+```@example model_accessing_basics
+reactions(sir)
+```
+We can count the number of reactions in a model using the `numreactions` function:
+```@example model_accessing_basics
+numreactions(sir)
+```
+Finally, a vector with all the reactions' rates can be retrieved using `reactionrates`:
+```@example model_accessing_basics
+reactionrates(sir)
+```
+
+### [Accessing content of models coupled to equations](@id model_accessing_basics_reactions)
+Previously, we have shown how to [couple equations to a chemical reaction network model](@ref ref), creating models containing [non-species unknowns (variables)](@ref ref). Here we create a birth-death model where some nutrient supply (modelled through the variable $N$) is depleted in the presence of $X$:
+```@example model_accessing_basics
+using Catalyst # hide
+coupled_crn = @reaction_network begin
+    @equations D(N) ~ -N * X
+    (p/(1+N),d), 0 <--> X
+end
+```
+Here, the `unknowns` function returns all unknowns (i.e. species and variables):
+```@example model_accessing_basics
+unknowns(coupled_crn)
+```
+Meanwhile, `species` returns the species only, while `nonspecies` returns the variables only:
+```@example model_accessing_basics
+species(coupled_crn)
+```
+```@example model_accessing_basics
+nonspecies(coupled_crn)
+```
+
+Similarly, the `equations` function returns a vector with all reactions and equations of the model (ordered so that reactions occur first and equations thereafter):
+```@example model_accessing_basics
+equations(coupled_crn)
+```
+Meanwhile, `reactions` returns the reactions only, while `nonreactions` returns any algebraic or differential equations:
+```@example model_accessing_basics
+reactions(coupled_crn)
+```
+```@example model_accessing_basics
+nonreactions(coupled_crn)
+```
+
+### [Accessing other model properties](@id model_accessing_basics_others)
+There exist several other functions for accessing model properties. 
+
+The `observed`, `continuous_events`, `discrete_events` functions can be used to access a model's [observables](@ref ref), [continuous events](ref ref), and [discrete events](@ref ref), respectively.
+
+The `ModelingToolkit.get_iv` function can be used to retrieve a [model's independent variable](@ref ref):
+```@example model_accessing_basics
+ModelingToolkit.get_iv(sir)
+```
+
+## [Accessing properties of hierarchical models](@id model_accessing_hierarchical)
```

Comment:
There is quite a lot on this, but I felt it was good to go through it properly (as I personally also have been a bit unfamiliar with it). Definitely check this part in case I have made a misstake.